### PR TITLE
Refactor thermal energy and heat capacities

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py [hunk 1]
```diff
@@ -9,40 +9,40 @@
 )
 
 # Description
-## Molar heat capacity of a substance is its heat capacity divided by its amount in moles.
+## Heat capacity is molar heat capacity (heat capacity per unit amount of substance) times amount of substance.
 
-# Definition: C_m = C / n
-## C_m - molar heat capacity
+# Definition: C = C_m * n
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py:14]: Now it is not a definition.
Also I disagree. It is more logical to define molar heat capacity from heat capacity.

File: examples/electricity/amount_energy_from_voltage_time_resistance.py [hunk 2]
```diff
@@ -27,45 +30,33 @@
 initial_temperature_kelvin = to_kelvin_quantity(initial_temperature)
 final_temperature_kelvin = to_kelvin_quantity(final_temperature)
 
-ohm_law_applied = ohm_law.law.subs({
-    ohm_law.current: operate_power.current,
-    ohm_law.resistance: joule_lenz_law.resistance,
-    ohm_law.voltage: joule_lenz_law.voltage
+water_mass = solve(
```
Comment (blackyblack) [inline examples/electricity/amount_energy_from_voltage_time_resistance.py:33]: Next time avoid doing major refactoring in examples and plots. They are merely for illustration purpose. We want many laws to work together in one example, not to have the most effective way of solving an example.

File: symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py (hunk 1 above)
Comment (alesanter) [reply symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py:14]: That's why I moved it to laws (although I forgot to rename to "Law" in the file.) We need to find the extensive heat capacity much more often than the intensive ones.
Plus, the intention behind this was to make the thermal energy law agnostic to the type of heat capacity and use other laws to substitute the correct one.

File: examples/electricity/amount_energy_from_voltage_time_resistance.py (hunk 2 above)
Comment (alesanter) [reply examples/electricity/amount_energy_from_voltage_time_resistance.py:33]: Well I wanted to get rid of the law that featured mass and specific heat capacity so on this account I needed to replace other usages of the former law.

File: symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py (hunk 1 above)
Comment (blackyblack) [reply symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py:14]: But we could derive this previous definition.

Comment (alesanter) [reply symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py:14]: You mean you want to have, say, both the `C = c * m` law and `c = C / m` definition?

Comment (blackyblack) [reply symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py:14]: No, I mean only definition. Law can be found by solving for C.

Comment (alesanter) [reply symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py:14]: So, okay, you'd prefer to have the definition `c = C / m` to having the law `C = c * m`, am I right?

Comment (blackyblack) [reply symplyphysics/laws/thermodynamics/heat_capacity_via_molar_heat_capacity.py:14]: Yes

File: plots/laws/thermodynamics/mass_of_ice_for_cooling_hot_water.py [hunk 3]
```diff
@@ -5,9 +5,11 @@
 from sympy.plotting.plot import MatplotlibBackend
 from symplyphysics import print_expression, global_index
 from symplyphysics.core.symbols.celsius import to_kelvin, Celsius
-from symplyphysics.laws.thermodynamics import thermal_energy_from_mass_and_temperature as energy_heating_law
+from symplyphysics.laws.thermodynamics import thermal_energy_from_heat_capacity_and_temperature as thermal_energy_law
 from symplyphysics.laws.thermodynamics import energy_to_melt_from_mass as energy_melting_law
 from symplyphysics.definitions import density_from_mass_volume as density_law
+# from symplyphysics.laws.thermodynamics import heat_capacity_via_specific_quantity as specific_qty_law
```
Comment (blackyblack) [inline plots/laws/thermodynamics/mass_of_ice_for_cooling_hot_water.py:11]: Delete.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**